lib/teamsnap.rb

```diff
@@ -31,19 +32,45 @@
 
 module TeamSnap
   EXCLUDED_RELS = ["me", "apiv2_root", "root", "self"]
-  DEFAULT_URL = "http://localhost:3000"
+  DEFAULT_URL = "http://apiv3.teamsnap.com"
   Error = Class.new(StandardError)
   NotFound = Class.new(TeamSnap::Error)
 
   class AuthMiddleware < Faraday::Middleware
     def initialize(app, options)
-      super(app)
       @options = options
+      super(app)
     end
 
     def call(env)
-      token = @options.fetch(:token)
-      env[:request_headers].merge!({"Authorization" => "Bearer #{token}"})
+      token = @options[:token]
+      client_id = @options[:client_id]
+      client_secret = @options[:client_secret]
+
+      if token
+        env[:request_headers].merge!({"Authorization" => "Bearer #{token}"})
+      elsif client_id && client_secret
+        query_params = Hash[URI.decode_www_form(env.url.query || "")]
+          .merge({
+          hmac_client_id: client_id,
+          hmac_nonce: SecureRandom.uuid,
+          hmac_timestamp: Time.now.to_i
+        })
+        env.url.query = URI.encode_www_form(query_params)
```

Comment:
fixed


*Shane Emmons* | Director of Technology and Principal Architect
TeamSnap - The easiest way to manage your team
(810) 834-9542 | shane@teamsnap.com | What is TeamSnap?
<http://www.youtube.com/watch?v=fx0vRXYGijU>

On Fri, Mar 6, 2015 at 2:41 PM, kristenboydstun <notifications@github.com>
wrote:

> In lib/teamsnap.rb
> <https://github.com/teamsnap/teamsnap_rb/pull/40#discussion_r25971652>:
>
> > -      token = @options.fetch(:token)
> > -      env[:request_headers].merge!({"Authorization" => "Bearer #{token}"})
> > +      token = @options[:token]
> > +      client_id = @options[:client_id]
> > +      client_secret = @options[:client_secret]
> > +
> > +      if token
> > +        env[:request_headers].merge!({"Authorization" => "Bearer #{token}"})
> > +      elsif client_id && client_secret
> > +        query_params = Hash[URI.decode_www_form(env.url.query || "")]
> > +          .merge({
> > +          hmac_client_id: client_id,
> > +          hmac_nonce: SecureRandom.uuid,
> > +          hmac_timestamp: Time.now.to_i
> > +        })
> > +        env.url.query = URI.encode_www_form(query_params)
>
> Indentation from lines 55-58 seems a little off
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/teamsnap/teamsnap_rb/pull/40/files#r25971652>.
>